pkg/networkservice/chains/nsmgr/server.go

```diff
@@ -83,8 +77,9 @@ var _ Nsmgr = (*nsmgrServer)(nil)
 func NewServer(ctx context.Context, nsmRegistration *registryapi.NetworkServiceEndpoint, authzServer networkservice.NetworkServiceServer, tokenGenerator token.GeneratorFunc, registryCC grpc.ClientConnInterface, clientDialOptions ...grpc.DialOption) Nsmgr {
 	rv := &nsmgrServer{}
 
-	var urlsRegistryServer registryapi.NetworkServiceEndpointRegistryServer
-	var localbypassRegistryServer registryapi.NetworkServiceEndpointRegistryServer
+	var setURLRegistryServer, interposeRegistryServer registryapi.NetworkServiceEndpointRegistryServer
+
+	filterMechanismsServer := filtermechanisms.NewServer(nsmRegistration.Url, &setURLRegistryServer)
```

Comment:
Why do we need to define this here? Can we move this to the chain at line 117?

Reply:
It fills `setURLRegistryServer` which is used in line 101.